proto/internal/temporal/server/api/matchingservice/v1/request_response.proto

```diff
@@ -355,15 +359,29 @@ message SyncDeploymentUserDataRequest {
 
     oneof operation {
         // The deployment version and its data that is being updated.
-        temporal.server.api.deployment.v1.DeploymentVersionData update_version_data = 6;
+        temporal.server.api.deployment.v1.DeploymentVersionData update_version_data = 6 [deprecated = true];
         // The version whose data should be cleaned from the task queue.
-        temporal.server.api.deployment.v1.WorkerDeploymentVersion forget_version = 7;
+        temporal.server.api.deployment.v1.WorkerDeploymentVersion forget_version = 7 [deprecated = true];
     }
+
+    // Absent means no change.
+    // Ignored by the task queue if new revision number is not greater that what it has.
+    temporal.api.deployment.v1.RoutingConfig update_routing_config = 10;
+    // Optional map of build id to upsert version data.
+    // Ignored if `update_routing_config` is present and has an outdated revision number.
```

Comment:
@Shivs11 probably best to ignore the whole request when it has out dated routing info. But that doesn't entirely protect us from applying out dated version data. I think we should be fine for now, but if that becomes an issue we can add separate counters for each version data too.